drivers/periph_common/spi.c

```diff
@@ -33,7 +33,7 @@ int spi_init_cs(spi_t bus, spi_cs_t cs)
     if (bus >= SPI_NUMOF) {
         return SPI_NODEV;
     }
-    if ((cs == SPI_CS_UNDEF) || (cs == GPIO_UNDEF)) {
+    if (gpio_is_equal(cs, SPI_CS_UNDEF) || !gpio_is_valid(cs)) {
```

Comment:
I understand that you want to keep this PR simply, basically just doing automatic conversion.
But mind you that `SPI_CS_UNDEF` is 

    #define SPI_CS_UNDEF    (GPIO_UNDEF)

Reply:
`SPI_CS_UNDEF` is overridable and might be something else than `GPIO_UNDEF`.

Reply:
Yea I just noticed that it's never overwritten.
So if we do away with `GPIO_UNDEF` this will probably go too.